osu.Game.Rulesets.Taiko/Mods/TaikoModSimplifiedRhythm.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Bindables;
+using osu.Framework.Localisation;
+using osu.Game.Beatmaps;
+using osu.Game.Beatmaps.ControlPoints;
+using osu.Game.Configuration;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Taiko.Beatmaps;
+using osu.Game.Rulesets.Taiko.Objects;
+
+namespace osu.Game.Rulesets.Taiko.Mods
+{
+    public class TaikoModSimplifiedRhythm : Mod, IApplicableToBeatmap
+    {
+        public override string Name => "Simplified Rhythm";
+        public override string Acronym => "SR";
+        public override double ScoreMultiplier => 0.6;
+        public override LocalisableString Description => "Simplify tricky rhythms!";
+        public override ModType Type => ModType.DifficultyReduction;
+
+        [SettingSource("One-third conversion", "Converts 1/3 snap to 1/2 snap.")]
+        public Bindable<bool> OneThirdConversion { get; } = new BindableBool(false);
+
+        [SettingSource("One-sixth conversion", "Converts 1/6 snap to 1/4 snap.")]
+        public Bindable<bool> OneSixthConversion { get; } = new BindableBool(true);
+
+        [SettingSource("One-eighth conversion", "Converts 1/8 snap to 1/4 snap.")]
+        public Bindable<bool> OneEighthConversion { get; } = new BindableBool(false);
+
+        public void ApplyToBeatmap(IBeatmap beatmap)
+        {
+            var taikoBeatmap = (TaikoBeatmap)beatmap;
+            var controlPointInfo = taikoBeatmap.ControlPointInfo;
+            List<Hit> toRemove = new List<Hit>();
+
+            // Snap conversions for rhythms
+            var snapConversions = new Dictionary<int, int>
+            {
+                { 8, 4 }, // 1/8 snap to 1/4 snap
+                { 6, 4 }, // 1/6 snap to 1/4 snap
+                { 3, 2 }, // 1/3 snap to 1/2 snap
+            };
+
+            int patternStartIndex = 0;
```

Comment:
Isolate the state here by moving it into `foreach (var snapConversion in snapConversions)`. Just to make sure the variable is reset at the beginning of each iteration.